sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java

```diff
@@ -798,6 +800,19 @@ public void shutdown() {
         suspend(false, new CancellationException());
     }
 
+    @Override
+    public void setNetworkRequests(boolean isAllowed) {
+        if (isAllowed) {
+            mCurrentState++;
+            for (GroupState groupState : mGroupStates.values()) {
+                checkPendingLogs(groupState);
+            }
+        } else  {
+            suspend(false, new CancellationException());
+            mEnabled = true;
```

Comment:
Could you please remove `mEnabled` changing from `suspend`, and change manually only where needed?